src/concrete/ml/sklearn/linear_model.py

```diff
@@ -421,26 +425,42 @@ def _fit_encrypted(
                 f"enabled. Got {y.shape}"
             )
 
+        if classes is not None and self.classes_ is not None:
+            if len(numpy.setxor1d(classes, self.classes_)) > 0:
+                raise ValueError(
+                    f"{classes=} is not the same as on last call to partial_fit,"
+                    f" was: {self.classes_}"
+                )
+
         # Build the quantized module
         # In case of a partial fit, only do so if it has not been done already (which indicates
         # that this is the partial fit's first call)
-        if not is_partial_fit or self.training_quantized_module is None:
+        if (not is_partial_fit) or (self.training_quantized_module is None):
             # Update this once we support multi-class
             # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/4182
             # We need to define this here and not in the init otherwise this breaks
             # because scikit-learn assumes that as soon as the attribute exists
             # the model is fitted
             # pylint: disable=attribute-defined-outside-init
-            self.label_encoder = LabelEncoder()
-            self.label_encoder.fit(y)
-            self.classes_ = self.label_encoder.classes_
+            if is_partial_fit and self.training_quantized_module is None and classes is None:
+                raise ValueError("classes must be passed on the first call to partial_fit.")
+
+            if classes is None:
+                self.label_encoder = LabelEncoder()
+                self.label_encoder.fit(y)
+                self.classes_ = numpy.array(self.label_encoder.classes_)
+            else:
+                self.label_encoder = LabelEncoder()
+                self.label_encoder.fit(classes)
+                self.classes_ = numpy.array(classes)
+            # If partial fit we should probably support adding classes along the way
 
             assert isinstance(self.classes_, numpy.ndarray)
 
             # Allow the training set to only provide a single class. This can happen, for example,
             # when running 'partial_fit' on a small batch of values. Even with a single class, the
             # model remains binary
-            if len(self.classes_) not in [1, 2]:
```

Comment:
not but this was a real feature, check the above comment : 
>             # Allow the training set to only provide a single class. This can happen, for example,
            # when running 'partial_fit' on a small batch of values. Even with a single class, the
            # model remains binary

basically, if the first `partial_fit` only has a single calss, we should support that since the model remains binary

Reply:
Actually the previous behavior was wrong. We still support partial fit with only one class in the training batch. In the case of calls to `partial_fit` the argument `classes` should be provided, as is the case in scikit-learn